mitodl/ocw-studio
# feat: Limit site starter options when creating new site

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-studio/issues/1354

#### What's this PR do?
- Adds a new dropdown/choices field named `status` in `websitestarter` model with values `Default`, `Active`, `Inactive`
- Updates the `websitestarter` list API to only include those starters which have status either `Default` or `Active` only.
- Shows `status` in website starters list display in Django admin. 
- While adding a new site,  shows the starter, by default, which has `status='Default'`. 
    - Note: Currently, multiple starters can have default status and only 1 of them (the top one) will be shown. Maybe we can restrict that only 1 starter can have a default status. https://github.com/mitodl/ocw-studio/issues/1354#issuecomment-1129829857

#### How should this be manually tested?
- Checkout this branch
- Apply migrations: `docker-compose run web python manage.py migrate`
- Verify that all your `websitestarters`, now have a status field, having `Active` as the default.
- Verify that this status is being shown in `websitestarters` list-display in Django admin.
- Go to the page of adding a new site:
    - Verify that all website starters are being shown in the starters dropdown.
    - Set a few statuses to `inactive` and now verify that these starters are now not showing in the list.
    - For any starter, set its status to `default`. Now verify that this starter is always selected/shown by default, whenever you add a new site, also verify that this site is created successfully.
- Any other use-case which needs to be tested.

#### Screenshots (if appropriate)
<img width="660" alt="image" src="https://user-images.githubusercontent.com/93309234/169015429-d4f54110-fa27-4683-99c4-5e94215a5dc4.png">

<img width="663" alt="image" src="https://user-images.githubusercontent.com/93309234/169015478-f90e2159-2499-4c42-96c7-15bc3167cef6.png">

<img width="817" alt="image" src="https://user-images.githubusercontent.com/93309234/169016616-79aeda61-16fb-4bf6-b90a-a8445109a516.png">

Starters with `default` and `active` status are shown only. Starter with `default` status is shown/selected by default when we open the add site page.

<img width="776" alt="image" src="https://user-images.githubusercontent.com/93309234/169016691-120667dd-1dc0-4c62-a0ef-12c8dff75403.png">

<img width="792" alt="image" src="https://user-images.githubusercontent.com/93309234/169016724-4a20bdf2-7fc4-4840-993d-49846cba421a.png">
